packages/app/src/specs/AverageDuration.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <div
+    v-if="props.gql.data?.__typename === 'CloudProjectSpec' && props.gql.data.averageDuration"
```

Comment:
There's value in leaving this condition inside `v-if`: it helps TSC to validate:
1. `props.gql.data` is not undefined
2. `props.gql.data.__typename` is `CloudProjectSpec`, which then allows us to access `averageDuration` when rendering content inside the div.

Refactoring to use a computed property will require we change line 7 to something like:
`{{ (props.gql.data?.__typename === 'CloudProjectSpec' && props.gql.data.averageDuration) ? getDurationString(props.gql.data.averageDuration ??) : '' }}`

Reply:
I may be missing a better approach, so please feel free to correct me.

Reply:
I agree with @mike-plummer , using named variables would increase the readability for other developers

> Refactoring to use a computed property will require we change line 7
I don't think we would need to refactor line 7 at all? But could be a Vue thing I'm unfamiliar with

Reply:
It is actually a TypeScript limitation. As I understand it, the TypeScript compiler and Vue tooling will evaluate `v-if` as if it was a regular `if` statement in code, meaning that the compiler will factor in the fact the condition must evaluate to `true` when evaluating the contents of the element with `v-if`.

This allows TypeScript to make the correct assumption that `props.gql.data` is a defined object with the shape correlated with `__typename === 'CloudProjectSpec'`, in contrast to other options in the union type where `__typename` is  `'CloudProjectSpecNotFound'` or `'CloudProjectUnauthorized'` which lack the definition of `averageDuration` field we're trying to access.

Maybe this screenshot can better show how the refactored code can throw off TSC
![image](https://user-images.githubusercontent.com/15657366/173624732-025b47bf-7d07-4fcc-9cb0-1b18dc632a8b.png)

For reference, this is the GQL union type definition:
https://github.com/cypress-io/cypress/blob/d80e719b32c5621f5ac79e92d200c789d3e529dc/packages/graphql/schemas/cloud.graphql#L278-L281

The corresponding generated TypeScript code (not checked in since it is generated) can be found in `packages/app/src/generated/graphql.ts` and here's a formatted version of it:

```ts
type AverageDurationFragment = {
  readonly __typename?: "RemoteFetchableCloudProjectSpecResult"
  readonly id: string
  readonly data:
    | {
        readonly __typename?: "CloudProjectSpec"
        readonly id: string
        readonly retrievedAt: string | null
        readonly averageDuration: number | null
      }
    | {
        readonly __typename?: "CloudProjectSpecNotFound"
        readonly retrievedAt: string | null
      }
    | { readonly __typename?: "CloudProjectUnauthorized" }
    | null
}

```

Reply:
TIL 😸 thank you for the detailed explanation!